feat: [CDS-73703]: Add ESO OCI Helm repository docs. (#2748)
harness/developer-hub: docs/continuous-delivery/gitops/add-a-harness-git-ops-repository.md

8. Select **Save & Continue**. Harness validates the connection.

:::note

Credentials Template is not supported for OCI Helm repository.

:::

```mdx-code-block
</TabItem>    
<TabItem value="OCI Helm repository with ESO" label="OCI Helm repository with ESO">
```

1. In **Specify Repository Type**, select **Helm**.
2. In **Repository Name**, enter a name.
3. In **GitOps Agent**, select or create the Agent you want to use to fetch charts from this repo. For details, go to [Install a Harness GitOps Agent](install-a-harness-git-ops-agent.md).
4. In **Repository URL**, enter the URL to your OCI Helm repository. For example, `us-east4-docker.pkg.dev/<gcp-project>/<repository>`.
5. Select **Enable OCI**
6. Select **Continue**.
7. In **Credentials**, select **Specify Credentials for Repository**.
    - Select the authentication option with **Username and password**, 

>       - for AWS you must put `AWS`, and for Google you put `oauth2accesstoken` for username.

Reviewer: ```suggestion
      - In the user name field, if you are authenticating to AWS, enter `AWS`. If you are authenticating to Google, enter `oauth2accesstoken`.
```
Is "username" the name of the field. Typically, we would begin the instruction with "In **User name**, enter the ..." Since I cannot see the UI now, I cannot tell whether the name of the field is "User name" or "Username" or something else. Please check and change my generic "In the user name field" to the typical wording I described above.

>       - for password you put your shortlived token. If you have [External Secrets Operator](https://docs.harness.io/article/3xqjzq2q2q-external-secrets-operator) **Refresh token** checkbox will appear, enable it.

Reviewer: ```suggestion
      - In **Password** enter your short-lived token. If you have an [External Secrets Operator](https://docs.harness.io/article/3xqjzq2q2q-external-secrets-operator), a **Refresh token** checkbox appears. Enable the checkbox.
```
Same as my previous comment, but I cannot think a password field would ever be called anything else. If I'm wrong, please check and correct.

>     - If you checked  **Refresh Token**, fill in  **Refresh Interval** (e.g. 1m, 1h, 12h, 1d). This is the interval at which the token will be refreshed.

Reviewer: ```suggestion
    - If you checked  **Refresh Token**, specify a  **Refresh Interval** (for example, 1m, 1h, 12h, or 1d). This is the interval with which you want the token to be refreshed.
```

    - Based on the url you entered, Harness will detect whether it is Google or AWS registry. And you can select type of authentication for the registry.
        - For Google, you can select **Google Service Account** or **Google Workload Identity**.

>           - If you select **Google Service Account**, you must upload the service [account key file](https://cloud.google.com/iam/docs/keys-create-delete). Paste the contents of file in the **Account Key** field. Contents of a file must be one line. For example:

Reviewer: ```suggestion
          - If you select **Google Service Account**, you must upload the service [account key file](https://cloud.google.com/iam/docs/keys-create-delete). Paste the contents of the file in the **Account Key** field. Contents of the file must be on a single line. For example:
```

          ```json
           { "type": "service_account", "project_id": "google-project-id", "private_key_id": "xxxx70c719xxxxbe7be090083xxxxxd85eca6", "private_key": "-----BEGIN PRIVATE KEY-----\nMIIEvgIBAD...ERhb7ipMxMQw6wpbF\n-----END PRIVATE KEY-----\n", .... "universe_domain": "googleapis.com" }
          ```

>           - If you select **Google Workload Identity**, you must enter the GCP Workload parameters. [Google Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity).

Reviewer: ```suggestion
          - If you select **Google Workload Identity**, you must enter the GCP Workload parameters. For more information, go to [Google Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity).
```

>           - Project ID is project where registry is located.

Reviewer: ```suggestion
          - Project ID is the project in which the registry is located.
```

        - For AWS, you can select **AWS Access Credentials** or **AWS Service Account**.

>           - If you select **AWS Access Credentials**, you must enter the AWS access key id and AWS secret access key, and optionally AWS session token. [AWS Credentials](https://docs.aws.amazon.com/general/latest/gr/aws-sec-cred-types.html#access-keys-and-secret-access-keys).

Reviewer: ```suggestion
          - If you select **AWS Access Credentials**, enter the AWS access key ID, the AWS secret access key, and, optionally, the AWS session token. For more information, go to [AWS Credentials](https://docs.aws.amazon.com/general/latest/gr/aws-sec-cred-types.html#access-keys-and-secret-access-keys).
```

>           - If you select **AWS Service Account**, you must enter the service account.

Reviewer: ```suggestion
          - If you select **AWS Service Account**, enter the service account.
```

>           - **Region** is the region where registry is located.

Reviewer: ```suggestion
          - **Region** is the region in which the registry is located.
```

>           - **Role** is the role arn that will be assumed with given credentials.

Reviewer: ```suggestion
          - **Role** is the role ARN that will be assumed with given credentials.
```
